src/realm/sync/protocol.cpp

```diff
@@ -247,6 +260,8 @@ Status protocol_error_to_status(ProtocolError error_code, std::string_view msg)
             case ProtocolError::user_blacklisted:
                 [[fallthrough]];
             case ProtocolError::transact_before_upload:
+                [[fallthrough]];
+            case ProtocolError::edge_reboot:
```

Comment:
This is very clearly not unreachable as we reach this point if the server sends us this error code. "Unreachable" specifically means that it's a bug in *our* code if things ever get to that point.